src/Dropdown/DropdownTriggers.js

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from "react";
+
+export default class ButtonTrigger extends Component {
+  constructor() {
+    super(...arguments);
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(event) {
+    this.props.onAction(event);
+  }
+
+  getSelectedHtml(item) {
+    if (item != null) {
+      return item.selectedHtml || item.html;
+    }
+
+    return null;
+  }
+
+  render() {
+    const { className, disabled, selectedItem } = this.props;
+
+    return (
+      <button
+        className={className}
+        disabled={disabled}
+        onClick={this.handleClick}
+        type="button"
+      >
+        {this.getSelectedHtml(selectedItem)}
+      </button>
+    );
+  }
+}
+
+export class InputTrigger extends ButtonTrigger {
```

Comment:
In my opinion having this extend `ButtonTrigger` only complicates things. The only "super" functionality it uses is `handleClick`.

Reply:
+1